<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>530</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ParamControl</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: #f5f5f5;</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <!-- Верхняя панель статуса и управления -->
    <item>
     <widget class="QWidget" name="statusPanel" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>50</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>50</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: #ffffff;</string>
      </property>
      <layout class="QHBoxLayout" name="statusLayout">
       <property name="leftMargin">
        <number>10</number>
       </property>
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="rightMargin">
        <number>10</number>
       </property>
       <property name="bottomMargin">
        <number>10</number>
       </property>
       <!-- Индикатор СОТМ -->
       <item>
        <widget class="QPushButton" name="connectionStatusButton">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>70</width>
           <height>30</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: IndianRed; border-color: IndianRed; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>СОТМ</string>
         </property>
        </widget>
       </item>
       
       <!-- Индикатор КА -->
       <item>
        <widget class="QPushButton" name="kaIndicatorButton">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: YellowGreen; border-color: YellowGreen; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>КА 101</string>
         </property>
        </widget>
       </item>
       
       <!-- Индикатор ЗС -->
       <item>
        <widget class="QPushButton" name="zsIndicatorButton">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: YellowGreen; border-color: YellowGreen; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>ЗС 123</string>
         </property>
        </widget>
       </item>
       
       <!-- Индикатор ТМИ -->
       <item>
        <widget class="QPushButton" name="tmiStatusButton">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: IndianRed; border-color: IndianRed; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>ТМИ</string>
         </property>
        </widget>
       </item>
       
       <!-- Индикатор СЕК -->
       <item>
        <widget class="QLabel" name="sekLabel">
         <property name="minimumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>60</width>
           <height>30</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: black; color: yellow; font-family: Consolas;</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="text">
          <string>СЕК:
--</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       
       <!-- Кнопка Старт -->
       <item>
        <widget class="QPushButton" name="startButton">
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>30</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: #3273dc; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>Старт</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/resources/icons/start.png</normaloff>:/resources/icons/start.png</iconset>
         </property>
        </widget>
       </item>
       
       <!-- Кнопка Стоп -->
       <item>
        <widget class="QPushButton" name="stopButton">
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>30</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: #e53e3e; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>Стоп</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/resources/icons/stop.png</normaloff>:/resources/icons/stop.png</iconset>
         </property>
        </widget>
       </item>
       
       <!-- Кнопка звука ТМИ -->
       <item>
        <widget class="QPushButton" name="tmiSoundToggleButton">
         <property name="minimumSize">
          <size>
           <width>130</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>130</width>
           <height>30</height>
          </size>
         </property>
         <property name="text">
          <string>Отключить звук ТМИ</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/resources/icons/sound_on.png</normaloff>:/resources/icons/sound_on.png</iconset>
         </property>
        </widget>
       </item>
       
       <!-- Скрытое текстовое поле для пути к звуковому файлу -->
       <item>
        <widget class="QLineEdit" name="textBoxNoTmiSound">
         <property name="visible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       
       <!-- Кнопка выбора звукового файла -->
       <item>
        <widget class="QPushButton" name="selectNoTmiSoundButton">
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Выбрать звук</string>
         </property>
        </widget>
       </item>
       
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       
       <!-- Кнопка настроек -->
       <item>
        <widget class="QToolButton" name="settingsButton">
         <property name="minimumSize">
          <size>
           <width>30</width>
           <height>30</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>30</width>
           <height>30</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: #4a5568; color: white; border-radius: 4px;</string>
         </property>
         <property name="text">
          <string>⚙</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/resources.qrc">
           <normaloff>:/resources/icons/settings.png</normaloff>:/resources/icons/settings.png</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    
    <!-- Основная область с параметрами и журналом -->
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      
      <!-- Левая панель: параметры -->
      <widget class="QWidget" name="parametersPanel">
       <layout class="QVBoxLayout" name="parametersLayout">
        <property name="leftMargin">
         <number>5</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QWidget" name="parametersTitleBar" native="true">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>25</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: #e2e8f0;</string>
          </property>
          <layout class="QHBoxLayout" name="parametersTitleLayout">
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="parametersTitle">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Параметры на контроле</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="addParameterButton">
             <property name="minimumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Добавить параметр</string>
             </property>
             <property name="text">
              <string>+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="removeParameterButton">
             <property name="minimumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Удалить параметр</string>
             </property>
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="editParameterButton">
             <property name="minimumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Редактировать параметр</string>
             </property>
             <property name="text">
              <string>ℹ</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTableView" name="parameterTableView">
          <property name="styleSheet">
           <string notr="true">background-color: #ffffff;</string>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="showGrid">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>100</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
      
      <!-- Правая панель: журнал -->
      <widget class="QWidget" name="logPanel">
       <layout class="QVBoxLayout" name="logLayout">
        <property name="leftMargin">
         <number>5</number>
        </property>
        <property name="topMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>5</number>
        </property>
        <property name="bottomMargin">
         <number>5</number>
        </property>
        <item>
         <widget class="QWidget" name="logTitleBar" native="true">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>25</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: #e2e8f0;</string>
          </property>
          <layout class="QHBoxLayout" name="logTitleLayout">
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="logTitle">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>События</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="showLogButton">
             <property name="minimumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Показать полный журнал</string>
             </property>
             <property name="text">
              <string>LOG</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="clearLogButton">
             <property name="minimumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>25</width>
               <height>25</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Очистить журнал</string>
             </property>
             <property name="text">
              <string>×</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTableView" name="logTableView">
          <property name="styleSheet">
           <string notr="true">background-color: #ffffff;</string>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="showGrid">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>100</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  
  <!-- Дополнительные кнопки для настройки соединения и т.д. -->
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomToolBarArea</set>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionConnectionSettings"/>
   <addaction name="actionChangeKaZs"/>
   <addaction name="separator"/>
   <addaction name="actionToggleParameterPanel"/>
  </widget>
  
  <action name="actionConnectionSettings">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/resources/icons/connect.png</normaloff>:/resources/icons/connect.png</iconset>
   </property>
   <property name="text">
    <string>Настройки соединения</string>
   </property>
   <property name="toolTip">
    <string>Настройки соединения с СОТМ</string>
   </property>
  </action>
  <action name="actionChangeKaZs">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/resources/icons/settings.png</normaloff>:/resources/icons/settings.png</iconset>
   </property>
   <property name="text">
    <string>Выбор КА и ЗС</string>
   </property>
   <property name="toolTip">
    <string>Выбор космического аппарата и земной станции</string>
   </property>
  </action>
  <action name="actionToggleParameterPanel">
   <property name="text">
    <string>&lt;&lt;&lt;</string>
   </property>
   <property name="toolTip">
    <string>Скрыть/показать панель параметров</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>